src/commands/cmd_zset.cc

```diff
@@ -221,64 +223,217 @@ class CommandZLexCount : public Commander {
   RangeLexSpec spec_;
 };
 
-class CommandZPop : public Commander {
+class CommandZPop : public Commander,
+                    private EvbufCallbackBase<CommandZPop, false>,
+                    private EventCallbackBase<CommandZPop> {
  public:
-  explicit CommandZPop(bool min) : min_(min) {}
+  explicit CommandZPop(bool min, bool block) : min_(min), block_(block) {}
```

Comment:
Sorry but, I actually can hardly find the benefit of merging the code for `BZPOP` and `ZPOP`. Most of I can see are lots of `if (block_)` checks to implement different logic separately, which seems just make the code more unreadable.

If you want to merge some thing to prevent duplication, I think you need to find some **common logic** so that these code can be treated uniformly rather than just put more `if`s, e.g.

```
f() { block a ... }
g() { block b ... }

// that is NOT a deduplication
h() { if(...) { block a ...} else { block b ... } }
```

Reply:
The differences between zpopmin/max and bzpopmin/max is that: 1.bzpopmin/max have an additional 'timeout' input parameter 2.bzpopmin/max have an additional 'key' output. The inner processing logic is the same.
In my first edition of the implementation, i implement the bzpopmin/max independent of zpopmin/max. But i received a comment that it has something duplicate. Maybe i misunderstood it. 
But i think it's Ok? There are also some 'block' in the implementation of bzmpop and zmpop.